Results_Engine/Query/MapResults.cs

```diff
@@ -72,22 +78,18 @@ public static partial class Query
             else
                 filteredRes = results;
 
-            Dictionary<string, IGrouping<string, TResult>> resGroups;
 
-            //Group results by Id and turn to dictionary
-            // Add null check for when the property of the name in whichId does not exist?
-            resGroups = filteredRes.GroupBy(x => Base.Query.PropertyValue(x, whichId).ToString()).ToDictionary(x => x.Key);
+            Func<TResult, string> resultIdFunction = GetResultIdentifier(results.First(), resultIdentifier);
+            //Turn to Lookup based on the result ID function
+            //Add null check for when the property of the name in whichId does not exist?
+            var resGroups = filteredRes.ToLookup(resultIdFunction);
```

Comment:
These could be turned again in a set of public overloaded Query methods for grouping results, which I think could come in handy in other cases. E.g.

GroupByIdentifier(results, resultIdentifier)